Replaces holo tape with holo barrier

phil235 opened the pull request:
Replaces holo tape and holo tape projector with holo barrier and holo barrier projector (children of the janitor's holosign creator).

All holosign_creators' code is in holosign_creator.dm and their icons in device.dmi
All holo signs themselves are in effects.dmi

The holo barrier is better than holotape in every way.
- It can be destroyed by all mob attacks and mech attacks, and projectiles.
- it's harder to abuse, you can't put more than one barrier on a floor tile
- you can't  put a barrier if there's something dense on the tile.
- Placing a holobarrier takes 3 seconds so you can't use it during chases.
- Holobarrier are placed one at a time so you can't quickly block a hallway.
- The barrier projector has a sign limit of 6 barriers to avoid abuse.
- You can't place barrier on windows. 
- Mobs with PASSTABLE, PASSGRILLE can pass through the barrier unrestricted, other mobs have to walk. 
- Things that pass through glass also pass the barrier.
- Projectile and thrown items pass over the barrier. 
- You can shoot the barrier. 
- The damage dealt to the barrier by hand are no longer random.

These changes also apply to the janitor's holosign projector (except creating a wet floor holosign is still instantaneous). The holo wet sign can be destroyed by attacks, and placing one does a click sound.

Shitty coder sprites courtesy of yours truly.

Fixes #7961

:cl: phil235
rscadd: Holo tape is replaced by holo barriers.
/:cl:

phil235 committed:
Replaces holo tape and holo tape projector with holo barrier and holo barrier projector (children of the janitor's holosign creator).

phil235 commented on code/game/objects/effects/portals.dm:
```diff
@@ -1,5 +1,6 @@
 
 /obj/effect
+	icon = 'icons/effects/effects.dmi'
```
so we don't have to add that line to more than half of /obj/effect's children.

phil235 commented on code/game/objects/items/holotape.dm:
```diff
@@ -1,254 +0,0 @@
-	if(!density) return 1
-	if(air_group || (height==0)) return 1
-
-	if((mover.flags & PASSGLASS || istype(mover, /obj/effect/meteor) || mover.throwing == 1) )
```
This line man.... PASSGLASS is a bit used by pass_flags, not flags. And a specific check for meteors but nothing for any projectile at all?

KazeEspada replied:
>Meteors go through. The one thing that an engineer would be most worried about.

Cheridan replied:
I'd bet anything there was some guy on Bay meticulously placing down holotape all round to block all meteors so they had to add this :P

phil235 committed:
Forgot to handle damage from thrown weapons.

KazeEspada commented on code/game/objects/items/weapons/holosign_creator.dm:
```diff
@@ -0,0 +1,189 @@
+/obj/effect/overlay/holograph/barrier/engineering
+	icon_state = "holosign_engi"
+
+/obj/item/weapon/caution
```
Why is this here? Its not holographic?

phil235 replied:
It was in the file janisigns.dm that I deleted and replaced by holosign_creator.dm
I didn't know where else to put it.

KazeEspada replied:
Miscenlaneous weapons?

Alek2ander commented:
Can mechs pass barriers?

KazeEspada commented:
No, they have to smash them. They are guarenteed to one hit smash them.

tkdrg commented on code/game/objects/items/weapons/holosign_creator.dm:
```diff
@@ -0,0 +1,189 @@
+/obj/item/weapon/holosign_creator
+	name = "holographic sign projector"
+	desc = "A handy-dandy hologaphic projector that displays a janitorial sign."
```
hologRaphic

tkdrg commented on code/game/objects/items/weapons/holosign_creator.dm:
```diff
@@ -0,0 +1,189 @@
+			if(!T.density) //can't put holograms on a tile that has dense stuff
+				for(var/atom/movable/AM in T)
+					if(AM.density)
+						return
```
you can replace this boilerplate with the `if(is_blocked_turf(T))` helper

tkdrg commented on code/game/objects/items/weapons/holosign_creator.dm:
```diff
@@ -0,0 +1,189 @@
+							return
+						for(var/atom/movable/AM in T) //don't try to sneak dense stuff on our tile during the wait.
+							if(AM.density)
+								return
```
can use the same helper here

tkdrg commented:
@phil235 looks fine to me but conflicts

phil235 committed:
Merge branch 'master' of https://github.com/tgstation/-tg-station into Holobarrier

Conflicts:
	icons/effects/effects.dmi

phil235 committed:
Moved janitorial sign in a new weapons/miscellaneous.dm
Made holosign_creator/afterattack() a bit better.
Fixed merge conflict fixing error.
Fixes typos in holosign_creator.dm
